patroni/utils.py

```diff
@@ -967,13 +975,18 @@ def cluster_as_json(cluster: 'Cluster') -> Dict[str, Any]:
         member.update({n: m.data[n] for n in optional_attributes if n in m.data})
 
         if m.name != leader_name:
-            lsn = m.lsn
-            if lsn is None:
-                member['lag'] = 'unknown'
-            elif cluster_lsn >= lsn:
-                member['lag'] = cluster_lsn - lsn
-            else:
-                member['lag'] = 0
+            for location in ('received_', 'replayed_', ''):
+                lsn_type, lag_type = f'{location}lsn', f'{location}lag'
+
+                lsn = getattr(m, lsn_type)
+                if lsn is None:
+                    member[lsn_type] = member[lag_type] = 'unknown'
```

Comment:
I wonder if we should set `*_lag` to `unknown`. It looks ugly at least in patronictl output, maybe at least in patronictl we should replace it with empty strings, because LSNs already say `unknown`.